salt/pillar/http_json.py

```diff
@@ -0,0 +1,52 @@
+# -*- coding: utf-8 -*-
+"""
+A module that adds data to the Pillar structure retrieved by an http request
+
+
+Configuring the HTTP_JSON ext_pillar
+====================================
+
+Set the following Salt config to setup Foreman as external pillar source:
+
+.. code-block:: json
+
+  ext_pillar:
+    - http_json:
+        url: http://example.com/api/minion_id
+        ::TODO::
+        username: username
+        password: password
+
+Module Documentation
+====================
+"""
+from __future__ import absolute_import
+
+# Import python libs
+import logging
+
+
+def ext_pillar(minion_id,
+               pillar,  # pylint: disable=W0613
+               url=None):
+    """
+    Read pillar data from HTTP response.
+
+    :param url String to make request
+    :returns dict with pillar data to add
+    :returns empty if error
+    """
+    # Set up logging
+    log = logging.getLogger(__name__)
```

Comment:
That makes sense. I will move it outside the function definition.
Oh this as well should I go and change the http_yaml.py as well in this same pull request?